### PR TITLE
Enable compression by default for Elasticsearch outputs

### PR DESCRIPTION
**Pull request by faec**
Change the default compression level for the Elasticsearch output from 0 to 1. Part of https://github.com/elastic/ingest-dev/issues/2458

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

The benchmarks evaluating this change were run by @alexsapran and discussed in https://github.com/elastic/ingest-dev/issues/2399. In summary we expect this to have the following effects on "typical" ingestion:

- ~70-80% reduction in data volume
- 20-25% CPU increase
- ~10% ingestion delay

The exact numbers depend on the specific configuration and the type of data being ingested. The CPU and ingestion changes were fairly stable in tests, and anything noticeably outside that range should be flagged for evaluation if it arises during testing.

### CODE REVIEW

**Comment by faec** [comment]
> I'm not too familiar with the monitoring output -- does compression also have to be set for the monitoring output?

I'm not aware of any discussion or testing of that case, so I wasn't planning to for this first pass. If we want to do that it should probably be included in the benchmarking and followup changes in https://github.com/elastic/ingest-dev/issues/2399.

**Comment by cmacknz** [comment]
> I'm not too familiar with the monitoring output -- does compression also have to be set for the monitoring output?

I believe this is only used for stack monitoring of standalone Beats and is never used when Beats are run under agent.

**Comment by strawgate** [comment]
The benchmark that shows a 25% CPU doesn't use any processors, right? When processors or more complex inputs are used we'd expect the % CPU impact to be lower (as the output represents a smaller total % of the CPU usage)?

**Review by leehinman — APPROVED** [review]

**Comment by faec** [comment]
> The benchmark that shows a 25% CPU doesn't use any processors, right? When processors or more complex inputs are used we'd expect the % CPU impact to be lower (as the output represents a smaller total % of the CPU usage)?

The quoted figures include some runs with processors (under Agent), but yes, configurations with heavy processor use or similar will see proportionately smaller marginal changes.

**Review by andrewkroh — APPROVED** [review]

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Review by belimawr — APPROVED** [review]

**Comment by cmacknz** [comment]
Merging, we have enough approvals and this change has been widely communicated.

**Comment by strawgate** [comment]
Can we backport the compression changes to latest 7.x?

**Comment by cmacknz** [comment]
Opened a 7.17 backport. Will wait for the final round of benchmark results before merging.

**Comment by nicpenning** [comment]
Any chance this change is causing a memory leak that is being reported across the community?

https://github.com/elastic/beats/issues/37142

**Comment by andrewkroh** [comment]
Have you tried setting `output.elasticsearch.compression_level: 0` to revert to the old behavior and seeing if the leak persists?

**Comment by nicpenning** [comment]
I am going to try that when I'm back at the office. I think others have more extreme examples that may be able to test more accurately but I will be happy to test and report back!

**Comment by hendry-lim** [comment]
> Have you tried setting `output.elasticsearch.compression_level: 0` to revert to the old behavior and seeing if the leak persists?

Yes, it persists. Metricbeat is consuming 4.4 GB of RAM now.

**Comment by cmacknz** [comment]
To close the loop the cause of the memory leak was fixed in https://github.com/elastic/elastic-agent-system-metrics/pull/115

**Pull request merged**
